packages/@aws-cdk/aws-ec2/lib/machine-image.ts

```diff
@@ -374,15 +381,24 @@ export class AmazonLinuxImage extends GenericSSMParameterImage {
    * Return the SSM parameter name that will contain the Amazon Linux image with the given attributes
    */
   public static ssmParameterName(props: AmazonLinuxImageProps = {}) {
-    const generation = (props && props.generation) || AmazonLinuxGeneration.AMAZON_LINUX;
-    const edition = (props && props.edition) || AmazonLinuxEdition.STANDARD;
-    const virtualization = (props && props.virtualization) || AmazonLinuxVirt.HVM;
-    const storage = (props && props.storage) || AmazonLinuxStorage.GENERAL_PURPOSE;
-    const cpu = (props && props.cpuType) || AmazonLinuxCpuType.X86_64;
+    let generation = (props && props.generation) || AmazonLinuxGeneration.AMAZON_LINUX;
+    let edition = (props && props.edition) || AmazonLinuxEdition.STANDARD;
+    let kernel = (props && props.kernel) || undefined;
+    let virtualization: AmazonLinuxVirt | undefined = (props && props.virtualization) || AmazonLinuxVirt.HVM;
+    let cpu = (props && props.cpuType) || AmazonLinuxCpuType.X86_64;
+    let storage: AmazonLinuxStorage | undefined = (props && props.storage) || AmazonLinuxStorage.GENERAL_PURPOSE;
+
+    if (generation === AmazonLinuxGeneration.AMAZON_LINUX_2022) {
+      kernel = AmazonLinuxKernel.KERNEL5_X;
+      storage = undefined;
+      virtualization = undefined;
```

Comment:
Setting these to undefined is one approach; the other would be to explicitly throw an error if the user has set them. The latter might yield less surprises down the road, when someone in the future is trying to use the hypothetical EBS-storage or PV-virtualized 2022 AL, but isn't getting the results they expect.

Reply:
```ts
let virtualization: AmazonLinuxVirt | undefined = (props && props.virtualization) || AmazonLinuxVirt.HVM;
...
let storage: AmazonLinuxStorage | undefined = (props && props.storage) || AmazonLinuxStorage.GENERAL_PURPOSE;
```

@njlynch  I like the approach of throwing an error. However, both `virtualization` and `storage` are set to `HVM` and `GENERAL_PURPOSE` above, thus always triggering the error when AL2022 is selected. That's the reason why I went with `undefined` approach for these two variables.  Are there any other ways to make the error throw approach happen?

Reply:
Something like this?

```
    let virtualization: AmazonLinuxVirt | undefined;
    let storage: AmazonLinuxStorage | undefined;

    if (generation === AmazonLinuxGeneration.AMAZON_LINUX_2022) {
      kernel = AmazonLinuxKernel.KERNEL5_X;
      if (props && props.storage) { } // throw error
      if (props && props.virtualization) {  } // throw error
    } else {
      storage = (props && props.storage) || AmazonLinuxStorage.GENERAL_PURPOSE;
      virtualization = (props && props.virtualization) || AmazonLinuxVirt.HVM;
    }
```

Reply:
Thank you!